src/providers/app-config/app-config.ts

```diff
@@ -88,18 +91,34 @@ export class AppConfigProvider {
 
   public loadRemoteConfig = (): Promise<any> =>
     this.getRemoteData()
+      .then((data: any) => {
+        const result: ValidationResult<any> = this.schemaValidatorProvider.validateRemoteConfig(data);
+
+        if (result.error === null) {
+          return data;
+        }
+
+        return Promise.reject(result.error);
```

Comment:
Personally, I would prefer the if statement to return the error and the return at the bottom to be the happy path. It being the other way round made it harder to read the first time

Reply:
Okay, good idea. Changing it